service/ValidationService.php

```diff
@@ -1,151 +1,147 @@
+        $queryParams = self::convertValidationParameters($validationParameters);
+
+        return NetLicensingService::getInstance()->post($context, $urlTemplate, $queryParams);
+    }
+
+    /**
+     * Perform validation without connecting to the server (offline) using validation file previously retrieved by
+     * {@link retrieveValidationFile}.
+     *
+     * @param Context $context determines the vendor on whose behalf the call is performed
+     *
+     * @param $validationFile string  validation file(response) returned by {@link retrieveValidationFile} call
+     *
+     * @param array $meta optional parameter, receiving messages returned within response <infos> section.
+     *
+     * @return ValidationResults|null result of the validation
+     * @throws BadSignatureException
+     * @throws Exception
+     */
+    static public function validateOffline(Context $context, $validationFile, array &$meta = [])
+    {
+        SignatureUtils::check($context, $validationFile);
+        return self::convertValidationResult($validationFile);
+    }
+
+    /**
+     * @param ValidationParameters $validationParameters
+     * @return array
+     */
+    static private function convertValidationParameters(ValidationParameters $validationParameters): array
+    {
+        $queryParams = [];
+
+        if ($validationParameters->getProductNumber()) {
+            $queryParams[Constants::PRODUCT_NUMBER] = $validationParameters->getProductNumber();
+        }
+
+        foreach ($validationParameters->getLicenseeProperties() as $key => $value) {
+            $queryParams[$key] = $value;
+        }
+
+        $pmIndex = 0;
+
+        foreach ($validationParameters->getParameters() as $productModuleName => $parameters) {
+            $queryParams[Constants::PRODUCT_MODULE_NUMBER . $pmIndex] = $productModuleName;
+            foreach ($parameters as $key => $value) {
+                $queryParams[$key . $pmIndex] = $value;
+            }
+            $pmIndex++;
+        }
+
+        return $queryParams;
+    }
+
+    /**
+     * @param $validationFile
+     * @param array $meta
+     * @return ValidationResults|null
+     * @throws Exception
+     */
+    static private function convertValidationResult($validationFile, array &$meta = [])
+    {
+        if (is_null($validationFile)) {
+            return null;
+        }
+
+        $validationResults = new ValidationResults();
+
+        if (!empty($validationFile->items->item)) {
+            foreach ($validationFile->items->item as $item) {
+                $array = ItemToArrayConverter::convert($item);
+                $validationResults->setProductModuleValidation($array[Constants::PRODUCT_MODULE_NUMBER], $array);
+            }
+
+            $validationResults->setTtl(new DateTime($validationFile->ttl));
+        }
+
+        if (!empty($validationFile->infos->info)) {
+            foreach ($validationFile->infos->info as $info) {
+                $meta[] = $info;

```

Comment:
Does this change fix another bug, not related to the ticket? If so, better doing it as a separate PR (for the future).